app/services/search_service.rb

```diff
@@ -0,0 +1,208 @@
+# The SearchService class is a utility class whose purpose is to provide detailed responses to queries within 
+# different categories (record types, functionality, subsystems, etc).
+# Though similar in operation to the TypeaheadService, the implementation is separate, in that the goal of the response
+# is to provide _detailed_ results at a deep level.  In effect, TypeaheadService provides pointers to 
+# better searches, while SearchService provides deep and detailed information.
+# TODO: Refactor TypeaheadService and SearchService so that common functions come from a higher level class?
+class SearchService
+
+  def initialize
+  end
+
+  def users(params)
+    @users ||= find_users(params)
+  end
+
+  def tags(params)
+    @tags ||= find_tags(params)
+  end
+
+  def notes(params)
+    @notes ||= find_notes(params)
+  end
+
+  def maps(params)
+    @maps ||= find_maps(params)
+  end
+
+  def comments
+    @comments ||= find_comments(params)
+  end
+
+  def find_users(input, limit=5)
+    DrupalUsers.limit(limit)
+        .order('uid DESC')
+        .where('name LIKE ? AND access != 0', '%' + input + '%')
+  end
+
+  def find_tags(input, limit=5)
+    DrupalTag.includes(:drupal_node)
+        .where('node.status = 1')
+        .limit(limit)
+        .where('name LIKE ?', '%' + input + '%')
+  end
+
+  def find_comments(input, limit=5)
+    DrupalComment.limit(limit)
+        .order('nid DESC')
+        .where('status = 1 AND comment LIKE ?', '%' + input + '%')
+  end
+
+  ## search for node title only
+  ## FIXme with solr
+  def find_notes(input, limit=5)
+    DrupalNode.limit(limit)
+        .order('nid DESC')
+        .where('type = "note" AND node.status = 1 AND title LIKE ?', '%' + input + '%')
+  end
+
+  def find_maps(input, limit=5)
+    DrupalNode.limit(limit)
+        .order('nid DESC')
+        .where('type = "map" AND node.status = 1 AND title LIKE ?', '%' + input + '%')
+  end
+  
+  # DEPRECATED
```

Comment:
Shall we have some idea of when we'd eliminate this -- say, when all referring code is eliminated, or something?